tests/api_internal/test_internal_api_call.py

```diff
@@ -0,0 +1,151 @@
+
+class TestInternalApiConfig(unittest.TestCase):
+    def setUp(self):
+        InternalApiConfig._initialized = False
+
+    @conf_vars(
+        {
+            ("core", "database_access_isolation"): "false",
+            ("core", "database_api_url"): "http://localhost:8888",
+        }
+    )
+    def test_get_use_internal_api_disabled(self):
+        self.assertFalse(InternalApiConfig.get_use_internal_api())
+
+    @conf_vars(
+        {
+            ("core", "database_access_isolation"): "true",
+            ("core", "database_api_url"): "http://localhost:8888",
+        }
+    )
+    def test_get_use_internal_api_enabled(self):
+        self.assertTrue(InternalApiConfig.get_use_internal_api())
+        self.assertEqual(
+            InternalApiConfig.get_internal_api_endpoint(),
+            "http://localhost:8888/internal/v1/rpcapi",
+        )
+
+
+@internal_api_call
+def fake_method() -> str:
+    return "local-call"
+
+
+@internal_api_call
+def fake_method_with_params(dag_id: str, task_id: int) -> str:
+    return f"local-call-with-params-{dag_id}-{task_id}"
+
+
+class TestInternalApiCall(unittest.TestCase):
+    def setUp(self):
+        InternalApiConfig._initialized = False
+
+    @conf_vars(
+        {
+            ("core", "database_access_isolation"): "false",
+            ("core", "database_api_url"): "http://localhost:8888",
+        }
+    )
+    @mock.patch("airflow.api_internal.internal_api_call.requests")
+    def test_local_call(self, mock_requests):
+        result = fake_method()
+
+        self.assertEqual(result, "local-call")
+        mock_requests.post.assert_not_called()
+
+    @conf_vars(
+        {
+            ("core", "database_access_isolation"): "true",
+            ("core", "database_api_url"): "http://localhost:8888",
+        }
+    )
+    @mock.patch("airflow.api_internal.internal_api_call.requests")
+    def test_remote_call(self, mock_requests):
+        response = requests.Response()
+        response.status_code = 200
+
+        response._content = json.dumps(BaseSerialization.serialize("remote-call"))
+
+        mock_requests.post.return_value = response
+
+        result = fake_method()
+        self.assertEqual(result, "remote-call")
+        expected_data = json.dumps(
+            {
+                "jsonrpc": "2.0",
+                "method": "tests.api_internal.test_internal_api_call.fake_method",
+                "params": '{"__var": {}, "__type": "dict"}',
+            }
+        )
+        mock_requests.post.assert_called_once_with(
+            url="http://localhost:8888/internal/v1/rpcapi",
+            data=expected_data,
+            headers={"Content-Type": "application/json"},
+        )
+
+    @conf_vars(
+        {
+            ("core", "database_access_isolation"): "true",
+            ("core", "database_api_url"): "http://localhost:8888",
+        }
+    )
+    @mock.patch("airflow.api_internal.internal_api_call.requests")
+    def test_remote_call_with_params(self, mock_requests):
```

Comment:
Same as my previous comment, this test can be merged with the previous test using `@pytest.mark.parametrize(`

Reply:
Do we really want to do it in this case? We call two different methods fake_method vs  fake_method_with_params, ad the assertion there are little big different

Reply:
It is a good point, I am fine keeping as is

Reply:
Yeah. let's keep them parameters. Sometimes coupling costs are bigger than DRY benefits (especially for tests DA DAMP is better than DRY).